support-services/build.sbt

```diff
@@ -13,5 +13,6 @@ libraryDependencies ++= Seq(
   "io.circe" %% "circe-generic" % circeVersion,
   "io.circe" %% "circe-generic-extras" % circeVersion,
   "io.circe" %% "circe-parser" % circeVersion,
-  "com.squareup.okhttp3" % "okhttp" % okhttpVersion
+  "com.squareup.okhttp3" % "okhttp" % okhttpVersion,
+  "com.gocardless" % "gocardless-pro" % "2.8.0"
```

Comment:
support-services sounds a bit like a "common" type module to me.  Every dependency that is added to this, will be on the classpath of everything that depends on this, meaning more risk of snyk issues, binary incompatibilities, increase in artifact size and difficult to reuse. The gravity of a "common" appears to be so strong, that over time it ends up as a monolithic everything library. (I learned all this from my membership-common experience)
I am sure that opinion varies on this, but I think having a "common" or "services" is an anti pattern for the above reason.
In terms of here, I would recommend creating a new gocardless sbt module, and put only the requrements/dependencies for that there.  Then it would be self contained and reusable independently.
@rupertbates probably has a useful viewpoint on this matter!

Reply:
fair points (including those mentioned in person) - since the new functions added to `GoCardlessService` aren't used in support-frontend and the old function isn't used by `support-workers` I'm going to create a sealed trait (or something) for `GoCardlessService` which resides in `support-services` but have the actual implementation**s** in the respective projects where the functions are used and so the GoCardless library dependency will be on only the necessary projects.

Reply:
done!

Reply:
I don't really agree with your concerns in this case John - there are 2 client apps for support-services and they both need the GoCardless library. I can't see any benefit from splitting that functionality between two separate projects, it's just duplication and runs the risk of things like the lib version getting updated in one project but not the other.

Reply:
sigh - i have now refactored - shall i refactor it back to shared, or shall i make a `val` for the version in `LibraryVersions.scala` (like `jacksonVersion` for example)

Reply:
Personally I'd go back to shared but it's up to you if you've got the patience!

Reply:
since there are only two clients it won't make any difference in practice at this stage, agreed.

Regarding the version going out of step, they are both in the same repo so we would have a common dependency version in Dependencies.scala.

As long as there will only ever be two clients, or if there were it would be obvious and easy to split the non-needed stuff out in some way, then that would be fine.  Perhaps in the case of membership-common there were other considerations (e.g. 2 complete changes of team members) that exacerbated it.  But the same happened on dotcom with the "common" module and I've seen it in many other projects.  I suppose it's a bit like microservices - split it down just enough, but not too much!  It's good that we're having the debate but maybe the specific line doesn't matter as much.